components/apiserver-proxy/cmd/proxy/main.go

```diff
@@ -221,11 +216,11 @@ func main() {
 	}
 
 	if cfg.insecureListenAddress != "" {
-		if cfg.upstreamForceH2C {
+		if cfg.upstreamForceH2C && !proxyForApiserver {
```

Comment:
Same here, if we are going to use it only for apiserver this whole if only clutters

Reply:
like above